test/unit/runner/providers/ProviderRegistryTest.php

```diff
@@ -0,0 +1,157 @@
+        ], [
+            'id'       => 'sockets',
+            'name'     => 'web sockets communicator',
+            'module'   => 'core/communicator/ws',
+            'bundle'   => 'loader/vendor.min',
+            'category' => 'communicator',
+            'active'   => true,
+            'tags'     => [ ]
+        ], [
+            'id'       => 'poll',
+            'name'     => 'poll communicator',
+            'module'   => "core/communicator/poll",
+            'bundle'   => 'loader/vendor.min',
+            'category' => 'communicator',
+            'active'   => true,
+            'tags'     => [ ]
+        ], [
+            'id' => 'qtiServiceProxy',
+            'module' => 'taoQtiTest/runner/proxy/qtiServiceProxy',
+            'bundle' => 'taoQtiTest/loader/qtiTestRunner.min',
+            'category' => 'proxy',
+            'active'   => true,
+            'tags'     => [ ]
+        ]
+    ];
+
+    /**
+     * Data provider
+     * @return array the data
+     */
+    public function categoryFindingProvider()
+    {
+        return [
+            [ null, [] ],
+            [ '', [] ],
+            [ 'foo', [] ],
+            [ 'runner', [self::$map[0], self::$map[1]] ],
+            [ 'communicator', [self::$map[2], self::$map[3],self::$map[4]]  ],
+            [ 'proxy', [self::$map[5]] ],
+        ];
+    }
+
+    /**
+     * Test getting providers by category
+     * @dataProvider categoryFindingProvider
+     */
+    public function testGetByCategory($category, $expectedProviders)
+    {
+        $registry = $this->getMockBuilder(ProviderRegistry::class)
+                ->setMethods(['getMap', 'setConfig'])
+                ->disableOriginalConstructor()
+                ->getMock();
+        $registry->method('getMap')->willReturn(self::$map);
+
+        $providers = $registry->getByCategory($category);
+        $this->assertEquals(array_values($providers), $expectedProviders);
+    }
+
+    /**
+     * Test removoing providers from a category
+     */
+    public function testRemoveByCategory()
+    {
+        $map = self::$map;
+        $registry = $this->getMockBuilder(ProviderRegistry::class)
+                ->setMethods(['getMap', 'setConfig','remove'])
+                ->disableOriginalConstructor()
+                ->getMock();
+
+        $registry->method('getMap')->willReturn($map);
+
+        $registry->expects($this->once())
+             ->method('remove')
+             ->with('taoQtiTest/runner/proxy/qtiServiceProxy');
+
+        $registry->removeByCategory('proxy');
+    }
+
+    /**
+     * Test removoing providers from a wrong category
+     */
+    public function testRemoveByWrongCategory()
+    {
+        $map = self::$map;
+        $registry = $this->getMockBuilder(ProviderRegistry::class)
+                ->setMethods(['getMap', 'setConfig','remove'])
+                ->disableOriginalConstructor()
+                ->getMock();
+
+        $registry->method('getMap')->willReturn($map);
```

Comment:
You can simply pass `self::$map` here and same in line 129 to produce less Opcode

Reply:
since `$map` is a copy of `self::$map` (not a reference, right ?) in that situation this was intentional to ensure the original is not modified